invenio_github/api.py

```diff
@@ -623,3 +625,20 @@ def resolve_record(self):
     def serialize_record(self):
         """Serializes the release record."""
         raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def badge_title(self):
+        """Stores a string to render in the record badge title (e.g. 'DOI')."""
+        return None
+
+    @property
+    @abstractmethod
+    def badge_value(self):
+        """Stores a string to render in the record badge value (e.g. '10.1234/invenio.1234')."""
+        return None
+
+    @property
+    def self_url(self):
+        """Release self url (e.g. github HTML url)."""
+        return self.release_payload.get("html_url")
```

Comment:
```suggestion
        raise NotImplementedError
```

minor: If we tie it to the record, then it should also be an `abstractmethod`